op-deployer/pkg/deployer/verify/etherscan_test.go

```diff
@@ -0,0 +1,279 @@
+package verify
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/time/rate"
+)
+
+const (
+	testAPIKey     = "test_api_key"
+	testAddressHex = "0x1234567890123456789012345678901234567890"
+	testTxHashHex  = "0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+	testGUID       = "verification_guid_12345"
+)
+
+// createTestClient creates a new EtherscanClient with a mock server for testing
+func createTestClient(t *testing.T, handler http.HandlerFunc) (*EtherscanClient, *httptest.Server) {
+	server := httptest.NewServer(handler)
+	t.Cleanup(func() { server.Close() })
+
+	// Use a fast rate limiter for testing
+	limiter := rate.NewLimiter(rate.Every(time.Millisecond), 10)
+	return NewEtherscanClient(testAPIKey, server.URL, limiter), server
+}
+
+// createTestArtifact creates a contract artifact for testing
+func createTestArtifact() *contractArtifact {
+	return &contractArtifact{
+		ContractName:    "TestContract",
+		CompilerVersion: "0.8.10",
+		EVMVersion:      "london",
+		Optimizer: OptimizerSettings{
+			Enabled: true,
+			Runs:    200,
+		},
+		Sources: map[string]SourceContent{
+			"TestContract.sol": {Content: "contract TestContract {}"},
+		},
+	}
+}
+
+func TestGetAPIEndpoint(t *testing.T) {
+	tests := []struct {
+		name          string
+		chainID       uint64
+		expected      string
+		expectedError bool
+	}{
+		{"Mainnet", 1, "https://api.etherscan.io/api", false},
+		{"Sepolia", 11155111, "https://api-sepolia.etherscan.io/api", false},
+		{"Unknown", 999, "", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := getAPIEndpoint(tt.chainID)
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expected, result)
+			}
+		})
+	}
+}
+
+func TestGetContractCreation(t *testing.T) {
+	client, _ := createTestClient(t, func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method)
```

Comment:
[nit] use `require` rather than assert

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/14633/commits/6a6d33d982437e0d806773f410cc7d1f6b7b1d61